### DISCUSSION TITLE
Installed LFS and I'm still getting the error message that a file is too large

### DISCUSSION THREAD

**Discussion by GraceEnsign** [community/community#36651] [Repositories]
Hello! I have installed LFS, did the 'git lfs track' for both a .csv and .json, and did the 'git add .gitattributes'. Two of my files (561kb and 13mb) pushed to the repository with no issue. But I have a json file that is 209mb that will not push.  I get the error 'this exceeds GitHub's file size limit of 100mb. You may want to try Git Large File Storage'. I'm not sure as to why the LFS is working for other files but not this one?

**Comment by ossanna16** [comment]
Hey there @GraceEnsign and thank you for asking a good question! I'm glad you found our community 🙂

**Reply by JulianHSC** [reply]
I'm also having the same problem. I saved the error message but I'm unsure if I should paste it here. For the record, I used Git Large File Storage - https://git-lfs.github.com.

**Comment by nepal143** [comment]
Hey there! I’ve worked extensively with Git LFS, and it sounds like you're almost there, but there might be a small misconfiguration with tracking your larger file. Here’s a step-by-step approach that should solve the issue:

First, confirm that Git LFS is actually tracking your large JSON file. Run this command to check:

``` git lfs ls-files ```

If your 209MB file doesn’t show up, that’s likely the problem.

If it's not tracked yet, you can explicitly add all .json files to LFS (or the specific one you need). Run:


``` git lfs track "*.json" ```


After tracking it, you need to re-add the file:

``` git add <path-to-your-json-file> ```

Double-check that the .gitattributes file reflects the tracking for your JSON file. You should see something like this in the file:

```  *.json filter=lfs diff=lfs merge=lfs -text ```

This ensures that all JSON files are being tracked by LFS.

If your large file was committed before setting up LFS, Git will still treat it as a regular file. You’ll need to unstage and recommit the file:

``` git rm --cached <path-to-your-json-file>
git add <path-to-your-json-file>
git commit -m "Re-added large JSON file with LFS" 
```

Finally, push the changes again:

```  git push origin main ``` 


Let me know how it goes, but this should cover all the bases! LFS works like a charm once everything is tracked properly. 😊

**Comment by ValentinoSpirito** [comment]
Hey Grace,

It sounds like you've done everything right with Git LFS, but that file size limit error can still be confusing.

GitHub’s standard upload limit for individual files is 100MB, even when using Git LFS. LFS helps manage large files more efficiently, but it doesn’t bypass GitHub’s hard cap on file size. That 209MB JSON file exceeds the limit, which is why you're getting the error, while the smaller files went through without issue.

One solution is to upgrade to GitHub Large File Storage (LFS) plan, which allows for larger files. Another option could be breaking that large JSON file into smaller chunks if feasible.

Let me know if you need more help figuring out the best path forward!

Best,
Valentino
